service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -190,8 +207,143 @@ public void prepareCustomDirections(Map<String, Object> data) {
         }
     }
 
-    public Map<String, Object> generateCMOTemplateData(Map<String, Object> caseData) {
-        return new HashMap<>();
+    // REFACTOR: 02/12/2019 Refactor this with CaseDataExtractionService and NotifyOfProceedingService to try and
+    //  extract common elements to CommonCaseDataExtractionService (maybe an separate service for docmosis templates?)
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> generateCMOTemplateData(Map<String, Object> caseDataMap) throws IOException {
```

Comment:
Same here, do not return map.

Reply:
we returned `Map` here as this is passed onto Docmosis render api

Reply:
[-1] I am OK (well not really but best what we can do right now) with map being returned.

However this class shouldn't work with the Map<String, Object> as input parameter we should work with our CaseData model immediately. I know you do conversion later, but I would much prefer strongly typed object being passed here.

Reply:
Altered case data so that it can now be passed